_delphi_utils_python/delphi_utils/backfill_profiler.py

```diff
@@ -0,0 +1,715 @@
+
+    filtered_backfill_df = backfill_df[backfill_df["lag"]<=max_lag]
+
+    pivot_df = pd.pivot_table(filtered_backfill_df,
+                              values="value",
+                              index=["geo_value", "time_value"],
+                              columns="lag").reset_index()
+    min_lag = filtered_backfill_df["lag"].min()
+    value_type = filtered_backfill_df["value_type"].values[0]
+    data_type = filtered_backfill_df["data_type"].values[0]
+
+    if not geo_values:
+        geo_values = pivot_df["geo_value"].unique()
+    max_lag =min(backfill_df["lag"].min(), max_lag)
+
+    if data_type == "completeness":
+        vmax = int((filtered_backfill_df["value"].max()//20 + 1) * 20)
+        vmin = int((filtered_backfill_df["value"].min()//20) * 20)
+        cbar_freq = int((vmax-vmin)/20)
+        cmap = "tab20c_r"
+        cbar_label = "Percentage"
+        ref_lag = backfill_df["ref_lag"].values[0]
+        fig_title = 'Percentage of Completion \
+                \nAgainst values reported %d days later\
+                \n%s %s, %s'%(ref_lag, source, value_type, "%s")
+    else:
+        vmax = ceil(filtered_backfill_df["value"].max())
+        vmin = floor(filtered_backfill_df["value"].min())
+        cbar_freq = int(vmax/20)
+        cmap = "YlGnBu"
+        cbar_label = "log10(Daily Change Rate)"
+        ref_lag = 0
+        fig_title = 'Daily Change Rate\n%s, %s, %s'%(source, value_type, "%s")
+
+    n_days = (end_date - start_date).days + 1 - ref_lag
+    time_index = np.array([(start_date + timedelta(i)).date() for i in range(n_days)])
+
+    for geo in geo_values:
+        sns.set(font_scale=1.2)
+        heatmap_df = pivot_df.loc[(pivot_df["geo_value"] == geo)
+                                  & (pivot_df["time_value"].isin(time_index)),
+                                  ["time_value"] + list(range(min_lag, max_lag+1))]
+        heatmap_df.set_index("time_value", inplace=True)
+        plt.figure(figsize = (18, 15))
+        plt.style.context("ggplot")
+        heatmap_df = heatmap_df.reindex(time_index)
+        selected_ytix = [x.weekday() == 6 for x in time_index] # Show Sundays on y_axis
+        ax = sns.heatmap(heatmap_df, annot=False, cmap=cmap, cbar=True,
+                         vmax=vmax, vmin=vmin, center=0,
+                         cbar_kws=dict(label=cbar_label,
+                                       ticks=list(range(0, vmax+1, cbar_freq))))
+        xtix = ax.get_xticks()
+        plt.xlabel("Lag", fontsize = 25)
+        plt.ylabel("Reference Date", fontsize = 25)
+        plt.title(fig_title%geo, fontsize = 30, loc="left")
+        plt.xticks(fontsize=20)
+        plt.yticks(fontsize=15)
+        ax.set_yticks(np.arange(0.5, n_days + 0.5)[selected_ytix])
+        ax.set_yticklabels(time_index[selected_ytix])
+        ax.set_xticks(xtix[::5])
+        plt.savefig(save_dir+"/"+str(geo)+".png")
+
+
+def create_lineplot_by_loations(save_dir:str, backfill_df:pd.DataFrame,
```

Comment:
```suggestion
def create_lineplot_by_locations(save_dir:str, backfill_df:pd.DataFrame,
```

Reply:
(so clearly this hasn't been called in tests -- is there any functionality in here that warrants testing?)